packages/embark/src/lib/modules/blockchain_process/blockchain.js

```diff
@@ -135,10 +139,13 @@ Blockchain.prototype.initStandaloneProcess = function () {
     // `embark run` without restarting `embark blockchain`)
     setInterval(() => {
```

Comment:
Rather than use `setInterval`, my preference would be for a recursive `setTimeout`. Reason: although unlikely, because the interval is 2000ms, there is a possibility of overlapped calls of `this.ipc.connect`, i.e. if for some reason the previous invocation's callback hasn't fired yet.

Reply:
That makes sense, agreed.